packages/app/pages/AllProjectsPage/AllProjectsPage.tsx

```diff
@@ -19,43 +21,140 @@ interface Props {
   children: Object
 }
 
+/**
+ * Row of buttons to filter which user agent types should be displayed
+ */
+class NavBar extends React.Component {
+
+  readonly navButtons = ["All", "Cooperative", "Projects", "Organizations", "Groups"]
+
+  private currentTheme;
+
+  constructor(private props) {
+    super(props)
+    this.state = {
+      activeButton: "All",
+    };
+
+    this.currentTheme = themeable(this.props.theme)
+  }
+
+  /**
+   * Handler function for when buttons are pressed.
+   * TODO apply the filter to the user agents
+   * @param buttonName the text of the button that was clicked
+   */
+  handleClick = (buttonName) => {
+    this.setState({ activeButton: buttonName })
+    this.forceUpdate(() => console.log('Finished Updating'))
+    // this.state.activeButton  = buttonName
+    console.log("Clicked", buttonName)
+    // TODO refresh the list and filter
+  };
+
+  /**
+   * Draws the filter buttons onto the filter bar
+   * @returns {any}
+   */
+  render() {
+    return (
+      <div {...this.currentTheme(2, "list_menu")}>
+        <ul id="project_type" {...this.currentTheme(3, "menu_type")}>
+
+          {this.navButtons.map((text, index) => (
+            <NavButton
+              i={index + 4}
+              text={text}
+              theme={this.props.theme}
+              onclick={this.handleClick}
+              active={this.state.activeButton === text}
+            />
+          ))}
+
+        </ul>
+        <div {...this.currentTheme(9, "menu_actions")}>
+          <button {...this.currentTheme(10, "actions_join")}>Create a new Project</button>
+        </div>
+
+      </div>
+    )
+  }
+}
+
+/**
+ * A single filter button. Includes handler and metadata to determine
+ * if it is currently active (or clicked)
+ */
+class NavButton extends React.Component {
+
+  currentTheme = themeable(this.props.theme)
 
+  render() {
+    return (
+      <li
+        onClick={() => {
+          console.log("Currently Active:", this.props.active)
+          this.props.onclick(this.props.text)
+          // this.props.state.activeButton = this.props.text
+        }}
+        {...this.currentTheme(this.props.i, "type_item", (this.props.active ? "active" : ""))}
+      >
+        {this.props.text}
+      </li>
+    )
+  }
+}
+
+/**
+ * A 'card' to display the user agent name, picture, and controls to
+ * join with the user agent.
+ */
+class ProjectCard extends React.Component {
+
+  currentTheme = themeable(this.props.theme)
+  i = this.props.i
+
+  render() {
+    return (
+      <div {...this.currentTheme((this.i * 7) + 13, "medium-6",  "end", "columns")}>
+        <div {...this.currentTheme((this.i * 7) + 14, "projects_item")}>
+          <div {...this.currentTheme((this.i * 7) + 15, "item_row")}>
+            <span {...this.currentTheme((this.i * 7) + 16, "row_image")}><img src={this.props.org.image} /></span>
```

Comment:
Avoid magic numbers and unnamed calculations

Reply:
This definitely needs to be better documented